pkg/v1/tkg/client/upgrade_cluster.go

```diff
@@ -442,6 +457,17 @@ func (c *TkgClient) applyPatchAndWait(regionalClusterClient, currentClusterClien
 	}
 	upgradeClusterConfig.UpgradeState = upgradeStateKCPPatchApplied
 
+	// If iaas == vsphere, get pod spec of kube-vip and change timeouts
+	kcp, err := regionalClusterClient.GetKCPObjectForCluster(upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace)
+	if err != nil {
+		return errors.Wrapf(err, "unable to find control plane node object for cluster %s", upgradeClusterConfig.ClusterName)
+	}
+	if kcp.Spec.MachineTemplate.InfrastructureRef.Kind == constants.VSphereMachineTemplate {
+		err = c.increaseKubeVipTimeouts(regionalClusterClient, upgradeClusterConfig)
+		if err != nil {
+			return errors.Wrap(err, "unable to increase kube-vip timeouts")
+		}
+	}
```

Comment:
I think this needs to be done before `patchKubernetesVersionToKubeadmControlPlane` to avoid cluster-api controller starting an upgrade with the old kubevip configuration and we can be on a safer side.